bin/router/src/shared_state.rs

```diff
@@ -8,12 +8,14 @@ use std::sync::Arc;
 
 use crate::jwt::JwtAuthRuntime;
 use crate::pipeline::cors::{CORSConfigError, Cors};
+use crate::pipeline::progressive_override::{OverrideLabelsCompileError, OverrideLabelsEvaluator};
 
 pub struct RouterSharedState {
     pub validation_plan: ValidationPlan,
     pub parse_cache: Cache<u64, Arc<graphql_parser::query::Document<'static, String>>>,
     pub router_config: Arc<HiveRouterConfig>,
     pub headers_plan: HeaderRulesPlan,
+    pub override_labels_evaluator: Arc<OverrideLabelsEvaluator>,
```

Comment:
As far as I can see from the code, this field doesn't need to be inside an `Arc` as the evaluator is not really shared by itself. The overall router state is shared (and Arc-ed) so you don't really need to `Arc` this one as well.